src/main/java/core/basesyntax/SalaryInfo.java

```diff
@@ -1,7 +1,96 @@
+
 public class SalaryInfo {
+    private static final DateTimeFormatter FORMATTER = DateTimeFormatter.ofPattern("dd.MM.yyyy");
+    private static final int DATE_INDEX = 0;
+    private static final int HOURS_INDEX = 2;
+    private static final int INCOME_INDEX = 3;
+    private static final int NAME_INDEX = 1;
+    private static final String DATA_DELIMITER = " ";
+    private static final String INDEX_NAME_DELIMITER = ":";
+    private static final String INDEX_NAME_PAIRS_DELIMITER = "|";
+    private static final String REPORT_DELIMITER = " - ";
+    private static final String REPORT_HEADER = "Report for period ";
+
     public String getSalaryInfo(String[] names, String[] data, String dateFrom, String dateTo) {
-        return null;
+        String indexedNameString = buildIndexedNameString(names);
+        int[] salaries = new int[names.length];
+        for (String dataRecord : data) {
+            String[] records = dataRecord.split(DATA_DELIMITER);
+            if (!indexedNameString.contains(records[NAME_INDEX])) {
+                continue;
+            }
+            LocalDate parsedRecordDate = parseStringToLocalDate(records[DATE_INDEX]);
+            if (parsedRecordDate.isBefore(parseStringToLocalDate(dateFrom))) {
+                continue;
+            }
+            if (parsedRecordDate.isAfter(parseStringToLocalDate(dateTo))) {
+                continue;
+            }
+            int salariesIndex = getSalariesIndexByName(indexedNameString, records[NAME_INDEX]);
+            try {
+                salaries[salariesIndex] +=
+                        Integer.parseInt(records[INCOME_INDEX]) * Integer.parseInt(
+                                records[HOURS_INDEX]);
+            } catch (NumberFormatException e) {
+                throw new RuntimeException(
+                        "Wrong format of working hours or income per hour in " + "records line.");
+            }
+        }
+        return composeReport(names, dateFrom, dateTo, salaries);
+    }
+
+    private static int getSalariesIndexByName(String indexedNameString, String employeeName) {
+        int startIndex = indexedNameString.lastIndexOf(INDEX_NAME_PAIRS_DELIMITER,
+                indexedNameString.indexOf(employeeName));
+        int endIndex = indexedNameString.indexOf(INDEX_NAME_DELIMITER, startIndex);
+        String indexSalaries = indexedNameString.substring(startIndex + 1, endIndex);
+        try {
+            return Integer.parseInt(indexSalaries);
+        } catch (NumberFormatException e) {
+            throw new RuntimeException("Wrong index format in indexed string of employee names.");
+        }
+    }
+
+    private static String composeReport(String[] names, String dateFrom, String dateTo,
+                                        int[] salaries) {
+        StringBuilder reportBuilder = new StringBuilder(REPORT_HEADER).append(dateFrom)
+                .append(REPORT_DELIMITER)
+                .append(dateTo)
```

Comment:
The 'REPORT_DELIMITER' constant is used to separate 'dateFrom' and 'dateTo' in the report. However, it is also hardcoded on line 66. To maintain consistency and avoid magic strings, you should use 'REPORT_DELIMITER' on line 66 as well.